toolkit/tools/pkg/imagecustomizerlib/customizeutils.go

```diff
@@ -206,6 +211,22 @@ func copyAdditionalFiles(baseConfigPath string, additionalFiles imagecustomizera
 	return nil
 }
 
+func copyAdditionalDirs(baseConfigPath string, additionalDirs imagecustomizerapi.DirConfigList, imageChroot *safechroot.Chroot) error {
+	for _, dirConfigElement := range additionalDirs {
+		absSourceDir := file.GetAbsPathWithBase(baseConfigPath, dirConfigElement.SourcePath)
+
+		logger.Log.Infof("Copying %s into %s", absSourceDir, dirConfigElement.DestinationPath)
+
+		err := imageChroot.AddDirs(absSourceDir, dirConfigElement.DestinationPath, fs.FileMode(*dirConfigElement.Permissions), fs.FileMode(*dirConfigElement.ChildPermissions))
```

Comment:
`Permissions` and `ChildPermissions` are optional fields. But you are assuming here that they are always provided.

Reply:
Now they are implemeneted as optional